packages/core/contracts/common/implementation/AncillaryData.sol

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+/**
+ * @title Library for encoding and decoding ancillary data for DVM price requests.
+ */
+library AncillaryData {
+    // TODO: Get source for this code, h/t Matt.
```

Comment:
https://ethereum.stackexchange.com/a/8447/47801

There's also a slightly different but more general implementation in the same thread: 

https://ethereum.stackexchange.com/a/58341/47801


Note: I think strings and bytes are roughly the same thing in solidity, but it might be worth checking that these implementations that were designed for strings also work for bytes (I believe the one you have, at least, does.